File: app/src/main/java/com/trevorwiebe/trackacow/adapters/ReportsRecyclerViewAdapter.java
package com.trevorwiebe.trackacow.adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.trevorwiebe.trackacow.R;
import com.trevorwiebe.trackacow.db.entities.LotEntity;

import java.util.ArrayList;


public class ReportsRecyclerViewAdapter extends RecyclerView.Adapter<ReportsRecyclerViewAdapter.ReportsViewHolder> {

    private ArrayList<LotEntity> mLotEntities = new ArrayList<>();

    private static final String TAG = "ReportsRecyclerViewAdap";

    public ReportsRecyclerViewAdapter() {
    }

    @Override
    public int getItemCount() {
        if (mLotEntities == null) return 0;
        return mLotEntities.size();
    }

    @NonNull
    @Override
    public ReportsViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.list_reports, viewGroup, false);
        return new ReportsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ReportsViewHolder reportsViewHolder, int i) {
        LotEntity lotEntity = mLotEntities.get(i);
        String lotName = lotEntity.getLotName();

        reportsViewHolder.mLotName.setText(lotName);
    }

    public void swapLotData(ArrayList<LotEntity> lotEntities) {
        mLotEntities = lotEntities;
        notifyDataSetChanged();
    }

    public class ReportsViewHolder extends RecyclerView.ViewHolder {

        private TextView mLotName;

        public ReportsViewHolder(View view) {
            super(view);
            mLotName = view.findViewById(R.id.reports_lot_name);
        }
    }
}

File: app/src/main/java/com/trevorwiebe/trackacow/domain/adapters/ReportsRecyclerViewAdapter.kt
package com.trevorwiebe.trackacow.domain.adapters

import androidx.recyclerview.widget.RecyclerView
import com.trevorwiebe.trackacow.domain.adapters.ReportsRecyclerViewAdapter.ReportsViewHolder
import com.trevorwiebe.trackacow.domain.models.lot.LotModel
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.trevorwiebe.trackacow.R
import android.widget.TextView

class ReportsRecyclerViewAdapter : RecyclerView.Adapter<ReportsViewHolder>() {

    private var mLotModel: List<LotModel> = emptyList()

    override fun getItemCount(): Int { return mLotModel.size }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ReportsViewHolder {
        val view = LayoutInflater.from(viewGroup.context).inflate(R.layout.list_reports, viewGroup, false)
        return ReportsViewHolder(view)
    }

    override fun onBindViewHolder(reportsViewHolder: ReportsViewHolder, i: Int) {
        val lotEntity: LotModel = mLotModel[i]
        val lotName = lotEntity.lotName
        reportsViewHolder.mLotName.text = lotName
    }

    fun swapLotData(lotModels: List<LotModel>) {
        mLotModel = lotModels
        notifyDataSetChanged()
    }

    inner class ReportsViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val mLotName: TextView = view.findViewById(R.id.reports_lot_name)
    }
}